src/coprocessor/endpoint.rs

```diff
@@ -425,69 +377,127 @@ impl Drop for RequestTracker {
     }
 }
 
-#[derive(Debug)]
 pub struct RequestTask {
-    req: Request,
-    cop_req: CopRequest,
-    ctx: ReqContext,
+    req_ctx: Arc<ReqContext>,
+    req_handler_builder: RequestHandlerBuilder,
```

Comment:
What's the benefit to store builder instead of CopRequest here?

Reply:
Here is the thing. As you can see, for each individual `ReqHandlers::new()` (for example, `DAGContext::new`), they accepts a `Snapshot`. However at time we build `RequestTask`, we don't have snapshots yet.. That's the reason why we have a `RequestHandlerBuilder`.